File: app/src/main/java/com/dertyp7214/appstore/adapter/UserAdapter.java
/*
 *
 *  * Copyright (c) 2018.
 *  * Created by Josua Lengwenath
 *
 */

/*
 * Copyright (c) 2018.
 * Created by Josua Lengwenath
 */

package com.dertyp7214.appstore.adapter;

import android.app.Activity;
import android.graphics.drawable.Drawable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.dertyp7214.appstore.R;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class UserAdapter extends RecyclerView.Adapter<UserAdapter.ViewHolder> {

    private Activity context;
    private List<User> users;
    private OnClick click;

    public UserAdapter(Activity context, List<User> users, OnClick click) {
        this.users = users;
        this.context = context;
        this.click = click;
    }

    @NonNull
    @Override
    public UserAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.user_view, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull UserAdapter.ViewHolder holder, int position) {
        User item = users.get(position);

        holder.img.setImageDrawable(item.getProfileImage());
        holder.img.setOnClickListener(click.onClick(item.getUid()));

    }

    public interface OnClick {
        View.OnClickListener onClick(String uid);
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        ImageView img;

        ViewHolder(View itemView) {
            super(itemView);

            img = itemView.findViewById(R.id.user_img);

        }
    }

    @Override
    public int getItemCount() {
        return users.size();
    }

    public static class User {
        private final String uid, name, email, created_at;
        private final Drawable profileImage;

        public User(String uid, String name, String email, String created_at, Drawable profileImage) {
            this.uid = uid;
            this.name = name;
            this.email = email;
            this.created_at = created_at;
            this.profileImage = profileImage;
        }

        public String getName() {
            return name;
        }

        public String getUid() {
            return uid;
        }

        public String getEmail() {
            return email;
        }

        public String getCreatedAt() {
            return created_at;
        }

        public Drawable getProfileImage() {
            return profileImage;
        }
    }
}

File: app/src/main/java/com/dertyp7214/appstore/adapter/UserAdapter.kt
/*
 *
 *  * Copyright (c) 2018.
 *  * Created by Josua Lengwenath
 *
 */

/*
 * Copyright (c) 2018.
 * Created by Josua Lengwenath
 */

package com.dertyp7214.appstore.adapter

import android.app.Activity
import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.dertyp7214.appstore.R

class UserAdapter(private val context: Activity, private val users: List<User>, private val click: OnClick) : RecyclerView.Adapter<UserAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserAdapter.ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context)
                .inflate(R.layout.user_view, parent, false))
    }

    override fun onBindViewHolder(holder: UserAdapter.ViewHolder, position: Int) {
        val item = users[position]

        holder.img.setImageDrawable(item.profileImage)
        holder.img.setOnClickListener(click.onClick(item.uid))
    }

    interface OnClick {
        fun onClick(uid: String): View.OnClickListener
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var img: ImageView = itemView.findViewById(R.id.user_img)
    }

    override fun getItemCount(): Int {
        return users.size
    }

    class User(val uid: String, val name: String, val email: String, val createdAt: String, val profileImage: Drawable)
}
